frontend/src/features/public-form/components/FormPaymentPage/components/ProductPaymentItemDetailsBlock.tsx

```diff
@@ -114,8 +114,19 @@ export const ProductPaymentItemDetailsBlock = ({
   paymentDetails: ProductsPaymentField
   colorTheme: FormColorTheme
 }) => {
-  const { register, setValue } = useFormContext()
-  register(PAYMENT_PRODUCT_FIELD_ID)
+  const {
+    register,
+    setValue,
+    formState: { errors },
+  } = useFormContext()
+  register(PAYMENT_PRODUCT_FIELD_ID, {
+    validate: (value) => {
+      const totalPrice = calculatePrice(value || [])
+      if (totalPrice === 0) {
```

Comment:
Hm. This has some additional black magic, and might be unexpected [1]. 

Would it be possible to calculate by products selected instead?

[1] Expectedness refers to assumptions by a lay person. I felt that a lay person would assume that the state of having something selected means that "at least 1 of the buttons needs to be selected" instead of basing off pricing (even though it is technically correct to make this assumption now).